pkg/logging/logging_test.go

```diff
@@ -136,12 +138,20 @@ func Test_withRotatingCore(t *testing.T) {
 			logger.Info("0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef")
 		}
 		assert.NoError(t, logger.Sync())
-		_, err := os.Stat(oldestRoll)
-		require.Error(t, err)
-		assert.Contains(t, err.Error(), oldestRoll+": no such file or directory", oldestRoll)
+		// lumberjack removes the old files asynchronously, therefore the test
+		// has to wait for it.
+		var err error
+		pause := 500 * time.Millisecond
```

Comment:
I would suggest to use [assert.EventuallyWithT](https://pkg.go.dev/github.com/stretchr/testify/assert#EventuallyWithT) here.